### PR TITLE
bau: add APM-server performance and error monitoring

### PR DESCRIPTION
**Pull request by tlwr**
What
----

Adds the ability to use an Elastic APM server to collect performance metrics and exceptions/traces.

How to review
-------------

- Code review
- Run APM server locally (or use https://govuk-paas-apm-server.london.cloudapps.digital) and run paas-admin locally with APM enabled

Who can review
---------------

Not @tlwr

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -1,4 +1,39 @@
 import pino from 'pino';
+const logger = pino({
+  level: process.env.LOG_LEVEL || 'info',
+});
+
+import apm from 'elastic-apm-node';
+/* istanbul ignore next */
+{
+  const isApmEnabled = !!process.env.ENABLE_APM;
```
Comment (richardTowers) [inline src/main.ts]: Could be a bit surprising that `ENABLE_APM=false ...` leads to `!!'false' === true`

Comment (tlwr) [reply src/main.ts]: Changed, thoughts?

**Comment by tlwr** [comment]
I'm marking this as not draft, I think we can get it on master and deployed without doing anything more

This will allow us to connect it to the live APM server temporarily without making changes to the pipeline.

File: src/main.ts [hunk 2]
```diff
@@ -1,4 +1,40 @@
...
+    throw new Error('APM_SERVER_URL is a required environment variable');
+  }
+
+  if (isApmEnabled) {
+    logger.info({
+      message: `Starting apm with name ${apmServiceName} -> ${apmServerUrl}`,
+    });
+  }
+
+  apm.start({
```
Comment (richardTowers) [inline src/main.ts]: Should this be in the `if (isApmEnabled)` block? or will it just do nothing if it's called with an empty URL

Comment (tlwr) [reply src/main.ts]: There is an `active` key in the object passed to start, so other bits of the code where we talk to `apm` don't have to be wrapped in an if statement

**Pull request merged**